utility/subcommands.py

```diff
@@ -0,0 +1,264 @@
+                            "output_hash": output_hash,
+                        }
+                    )
+                    existing_hashes.add(output_hash)
+            current_command, current_output_lines = cmd_match.group(0).strip(), []
+        else:
+            current_output_lines.append(cleaned_line)
+
+    return {"outputs": results}
+
+
+def save_to_remote(command, output, subcomponent_filter, output_directory):
+    """
+    Saves extracted command outputs to structured JSON files, avoiding duplicates.
+    Args:
+        command (str): The extracted `radosgw-admin` command.
+        output (dict): The parsed output of the command.
+        subcomponent_filter (str): The specified subcomponent filter.
+        output_directory (str): The directory where JSON files will be saved.
+    """
+    output_hash = compute_output_hash(output)
+    base_dir = os.path.join(output_directory, "cephci-commands-results")
+    os.makedirs(base_dir, exist_ok=True)
+
+    subcommand_match = re.search(r"radosgw-admin (\w+)", command)
+    if subcommand_match:
+        subcommand = subcommand_match.group(1)
+        file_path = os.path.join(base_dir, f"{subcommand}_outputs.json")
+
+        try:
+            if not os.path.exists(file_path):
+                with open(file_path, "w") as file:
+                    json.dump({"outputs": []}, file)
+            with open(file_path, "r") as file:
+                data = json.load(file)
+            if not any(
+                entry["output_hash"] == output_hash for entry in data["outputs"]
+            ):
+                data["outputs"].append(
+                    {"command": command, "output": output, "output_hash": output_hash}
+                )
+                with open(file_path, "w") as file:
+                    json.dump(data, file, indent=4)
+        except Exception as e:
+            print(f"Error saving file: {e}")
+
+
+def get_log_files_from_directory(directory):
+    """
+    Retrieves all `.log` files from a given directory.
+    Args:
+        directory (str): The directory to scan.
+    Returns:
+        list: List of file paths to `.log` files.
+    """
+    if not isinstance(directory, str):
+        raise TypeError("Expected a string path for directory")
+    return [
+        os.path.join(root, file)
+        for root, _, files in os.walk(directory)
+        for file in files
+        if file.endswith(".log")
+    ]
+
+
+def process_log_file(file_path, subcomponent_filter, output_directory):
+    """
+    Processes a single log file for command extraction and saves the outputs.
+    Args:
+        file_path (str): The path to the log file.
+        subcomponent_filter (str): The subcomponent filter.
+        output_directory (str): The directory where extracted data will be saved.
+    """
+    try:
+        with open(file_path, "r") as file:
+            log_lines = file.readlines()
+            extracted_data = extract_radosgw_admin_commands(log_lines)
```

Comment:
This becomes `component` specific. IMO, we should check if it is `rgw` component and then only call this method.